File: tests/fixtures/repomap/sample.kt
import kotlin.collections.List
import kotlin.io.println

class UserService(private val db: Database) {
    fun getUser(id: Int): User? {
        return db.find(id)
    }

    fun listUsers(): List<User> {
        return db.all()
    }

    private fun validate(id: Int): Boolean {
        return id > 0
    }
}

object AppConfig {
    val port = 8080
    val host = "localhost"
}

fun createApp(config: AppConfig): UserService {
    return UserService(Database())
}

data class User(val id: Int, val name: String)

File: tests/fixtures/repomap/sample.scala
import scala.collection.mutable
import scala.io.Source

class UserService(db: Database) {
  def getUser(id: Int): Option[User] = {
    db.find(id)
  }

  def listUsers(): List[User] = {
    db.all()
  }
}

object AppConfig {
  val port: Int = 8080
  val host: String = "localhost"
}

trait Repository {
  def save(entity: Any): Unit
  def find(id: Int): Option[Any]
}

def createApp(config: AppConfig): UserService = {
  new UserService(new Database())
}

val defaultTimeout: Int = 30
